src/Components/WorkloadRules/WorkloadRules.js

```diff
@@ -104,48 +134,158 @@ const WorkloadRules = ({ workload }) => {
   };
 
   const buildFilteredRows = (allRows, filters) => {
-    void filters;
     setRowsFiltered(false);
+    const noFilters = filtersAreApplied(filters);
+    return allRows
+      .filter((recs) =>
+        noFilters ? passFilterWorkloadsRecs(recs, filters) : true
+      )
+      .map((value, key) => [
+        {
+          rule: value,
+          isOpen: isAllExpanded,
+          cells: [
+            {
+              title: value.details,
+            },
+            {
+              title: (
+                <div key={key}>
+                  <InsightsLabel value={4} rest={{ isCompact: true }} />
+                </div>
+              ),
+            },
+            {
+              title: value.objects.length,
+            },
+            {
+              title: (
+                <div key={key}>
+                  <DateFormat date={value.modified} type="relative" />
+                </div>
+              ),
+            },
+          ],
+        },
+        {
+          cells: [
+            {
+              title: (
+                <ExpandedRulesDetails
+                  extra_data={value.extra_data}
+                  more_info={value.more_info}
+                  resolution={value.resolution}
+                  objects={value.objects}
+                />
+              ),
+            },
+          ],
+        },
+      ]);
+  };
 
-    return allRows.map((value, key) => [
-      {
-        rule: value,
-        isOpen: isAllExpanded,
-        cells: [
-          {
-            title: value.description,
-          },
-          {
-            title: (
-              <div key={key}>
-                <InsightsLabel value={4} rest={{ isCompact: true }} />
-              </div>
-            ),
-          },
-          {
-            title: value.objects.length,
-          },
-          {
-            title: (
-              <div key={key}>
-                {/* <DateFormat
-                  date={value.created_at}
-                  type="relative"
-                  tooltipProps={{ position: TooltipPosition.bottom }}
-                /> */}
-              </div>
-            ),
-          },
-        ],
-      },
-      {
-        cells: [
-          {
-            title: <ExpandedRulesDetails recommendations={recommendations} />,
-          },
-        ],
-      },
-    ]);
+  const pruneFilters = (localFilters, filterCategories) => {
+    const prunedFilters = Object.entries(localFilters);
+    return prunedFilters.length > 0
+      ? prunedFilters.reduce((arr, item) => {
```

Comment:
nit-pick (for better readability):
```suggestion
      ? prunedFilters.reduce((arr, [name, value]) => {
```